### PR TITLE
initial version of cmake config

### PR DESCRIPTION
**Pull request by grdanny**
adding a CMake option for building

### CODE REVIEW

**Comment by grdanny** [comment]
To build, from the root of the repo do:

$ mkdir build
$ cd build
$ cmake -G "Unix Makefiles" ..
$ cmake --build . --config Release --target pystring

If you want to also run the test app:
$ cmake --build . --config Release --target pystring_test
$ ctest

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,22 @@
+cmake_minimum_required(VERSION 2.8.12)
```
Comment (lgritz) [inline CMakeLists.txt:1]: I would recommend a 3.x minimum (3.13? 3.15?).

**Review by lgritz — APPROVED** [review]
LGTM, there's nothing wrong with this as a start.

I recommend a higher minimum, in order to encourage any further improvements to the file to take advantage of modern cmake idioms.

An obvious place for improvement in a subsequent change would be to export a pystringConfig.cmake that downstream projects can use.

**Comment by lgritz** [comment]
LGTM

**Pull request merged**